tachyon/math/polynomials/univariate/lagrange_interpolation.h

```diff
@@ -0,0 +1,68 @@
+#ifndef TACHYON_MATH_POLYNOMIALS_UNIVARIATE_LAGRANGE_INTERPOLATION_H_
+#define TACHYON_MATH_POLYNOMIALS_UNIVARIATE_LAGRANGE_INTERPOLATION_H_
+
+#include <utility>
+#include <vector>
+
+#include "tachyon/math/polynomials/univariate/univariate_polynomial.h"
+
+namespace tachyon::math {
+
+template <typename F, size_t MaxDegree>
+UnivariateDensePolynomial<F, MaxDegree> LagrangeInterpolate(
+    std::vector<F> points, std::vector<F> evals) {
+  using Poly = UnivariateDensePolynomial<F, MaxDegree>;
+  using Coeffs = UnivariateDenseCoefficients<F, MaxDegree>;
+
+  CHECK_EQ(points.size(), evals.size());
+  CHECK_EQ(points.size(), MaxDegree + 1);
+
+  if (points.size() == 1) {
+    return Poly(Coeffs({evals[0]}));
+  }
+
+  // points = [x₀, x₁, ..., xₙ]
+  // denoms[j][k] = (xⱼ - xₖ)
```

Comment:
Please use the same letter below `domains[i][j]`. By the way, I prefer looping with i,j and k...